controller/dataplane/watch.go

```diff
@@ -27,5 +37,64 @@ func DataPlaneWatchBuilder(mgr ctrl.Manager) *builder.Builder {
 		// Watch for changes in HPA created by the dataplane controller.
 		Owns(&autoscalingv2.HorizontalPodAutoscaler{}).
 		// Watch for changes in PodDisruptionBudgets created by the dataplane controller.
-		Owns(&policyv1.PodDisruptionBudget{})
+		Owns(&policyv1.PodDisruptionBudget{}).
+		// Watch for changes in ConfigMaps created by the dataplane controller.
+		Owns(&corev1.ConfigMap{}).
+		// Watch for changes in ConfigMaps that are mapped to KongPluginInstallation objects.
+		// They may trigger reconciliation of DataPlane resources.
+		Watches(
+			&corev1.ConfigMap{},
+			handler.EnqueueRequestsFromMapFunc(listDataPlanesReferencingKongPluginInstallation(mgr.GetClient())),
+		)
+}
+
+func listDataPlanesReferencingKongPluginInstallation(
+	c client.Client,
+) handler.TypedMapFunc[client.Object, reconcile.Request] {
```

Comment:
I tried to do it and failed using `Watches(...)`. I think it can be done only in the way I did based on what you provided me, please check

Reply:
👍 

Just an FYI: builder's `Watches()` can't be easily done now given that go doesn't allow type parameters on methods: https://github.com/golang/go/issues/49085

I've created this to track the issue (but knowing that it can't be easily done now). https://github.com/kubernetes-sigs/controller-runtime/issues/2955.

So your approach with using source seems to be the only one that's possible.